agent/tool-scripts/external-data-source

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: t; sh-basic-offset: 8; sh-indentation: 8; sh-indent-for-case-alt: + -*-
+
+script_path=`dirname $0`
+script_name=`basename $0`
+pbench_bin="`cd ${script_path}/..; /bin/pwd`"
+
+# source the base script
+. "$pbench_bin"/base
+
+# Perftool scripts must provide the following functions
+# 1) Install the tool
+# 2) Start data collection
+# 3) Stop data collection
+# 4) post-process the data
+
+# Defaults
+tool=$script_name
+tool_package_name="pbench-sysstat"
+tool_package_ver="11.2.0"
+tool_bin=/usr/local/bin/$tool
+group=default
+dir="/tmp"
+mode=""
+interval="10"
+iteration="1"
+options=""
+
+# Process options and arguments
+opts=$(getopt -q -o idp --longoptions "type:,url:,group:,options:,start,stop,install,postprocess" -n "getopt.sh" -- "$@");
+if [ $? -ne 0 ]; then
+	printf "\n"
+	printf "$script_name: you specified an invalid option\n\n"
+	printf "The following options are available: \n\n"
+	printf -- "\t--install,			install this perf tool\n"
+	printf "\n"
+	printf -- "\t--start|stop|postprocess	start/stop/post-process the data collection\n"
+       printf -- "\t--type=str 		the type (required)\n"
+       printf -- "\t--url=str 		the path of the data source (required)\n"
+	printf -- "\t--group=str		the perftool group (required)\n"
+	printf -- "\t--options=str		options passed directly to the tool\n"
+	exit 1
+fi
+eval set -- "$opts";
+while true; do
+	case "$1" in
+		--install)
+		mode="install"
+		shift;
+		;;
+		--start)
+		mode="start"
+		shift;
+		;;
+		--stop)
+		mode="stop"
+		shift;
+		;;
+		--postprocess)
+		mode="postprocess"
+		shift;
+		;;
+		-g|--group)
+		shift;
+		if [ -n "$1" ]; then
+			group="$1"
+			shift
+		fi
+		;;
+                --type)
+                shift;
+                if [ -n "$1" ]; then
+                        type="$1"
+			shift
+		fi
+		;;
+		--url)
+                shift;
+                if [ -n "$1" ]; then
+                        url="$1"
+                        shift
+                fi
+                ;;
+		-o|--options)
+		shift;
+		if [ -n "$1" ]; then
+			options="$1"
+			shift
+		fi
+		;;
+		--)
+		shift;
+		break;
+		;;
+	esac
+done
+
+tool_dir="$dir/tools-$group"
+tool_output_dir=$tool_dir/$tool # all tools keep data in their tool specific dir
+tool_cmd_file="$tool_output_dir/$tool.cmd"
+tool_pid_file=$pbench_tmp/$group.$iteration.$tool.pid
+tool_stdout_file=$tool_output_dir/$tool-stdout.txt
+tool_stderr_file=$tool_output_dir/$tool-stderr.txt
+case "$mode" in
+        install)
+        ;;
+        start)
+        ;;
+        stop)
+        ;;
+	postprocess)
```

Comment:
Since metadata about tools is already captured and indexed into Elasticsearch, and that metadata includes all the options given to tools, having a post-processing step seems redundant.  Can we remove this script entirely and make it a no-op like the other options?